File: legacy/node/src/commands/status.js
const chalk = require('chalk');
const { loadJSON, CONFIG_PATH, HISTORY_PATH, ASSIGNMENTS_PATH, TOTEM_PATH } = require('../utils/fileManager');
const { getTodayDate } = require('../utils/dateHelper');
const { normalizeTotem } = require('../utils/totemManager');

// Commande: Afficher le statut
function showStatus() {
  const config = loadJSON(CONFIG_PATH);
  const history = loadJSON(HISTORY_PATH);
  const assignments = loadJSON(ASSIGNMENTS_PATH);
  
  console.log(chalk.bold.cyan('\n📊 Statut des corvées\n'));
  
  // Afficher l'historique total
  console.log(chalk.bold('Historique total :'));
  Object.entries(history)
    .sort((a, b) => a[1] - b[1])
    .forEach(([user, count]) => {
      const bar = '█'.repeat(count);
      console.log(chalk.yellow(user.padEnd(30)), chalk.cyan(bar), chalk.gray(`(${count})`));
    });
  
  console.log('');
  
  // Afficher les dernières attributions
  const today = getTodayDate();
  const todayAssignments = assignments.filter(a => a.date === today);
  
  if (todayAssignments.length > 0) {
    console.log(chalk.bold(`Attributions du jour (${today}) :`));
    todayAssignments.forEach(assignment => {
      const status = assignment.completed ? chalk.green('✓') : chalk.gray('○');
      console.log(status, chalk.bold(assignment.chore.padEnd(15)), '→', chalk.yellow(assignment.user));
    });
  } else {
    console.log(chalk.gray('Aucune attribution pour aujourd\'hui.'));
  }

  const totemRaw = loadJSON(TOTEM_PATH);
  const totem = normalizeTotem(totemRaw, config.users);
  console.log(chalk.bold('\nTotem d\'immunité :'));
  if (totem.immune.length === 0) {
    console.log(chalk.gray('  Immunisés : aucun'));
  } else {
    console.log(chalk.gray('  Immunisés :'), chalk.yellow(totem.immune.join(', ')));
  }
  if (totem.safe.length === 0) {
    console.log(chalk.gray('  À l\'abri : aucun'));
  } else {
    console.log(chalk.gray('  À l\'abri :'), chalk.cyan(totem.safe.join(', ')));
  }
  if (totem.forcedQueue.length === 0) {
    console.log(chalk.gray('  File forcée : aucune'));
  } else {
    console.log(chalk.gray('  File forcée :'), chalk.cyan(totem.forcedQueue.join(', ')));
  }
  
  // Afficher les 5 dernières attributions (hors aujourd'hui)
  const recentAssignments = assignments
    .filter(a => a.date !== today)
    .slice(-5)
    .reverse();
  
  if (recentAssignments.length > 0) {
    console.log(chalk.bold('\nDernières attributions :'));
    recentAssignments.forEach(assignment => {
      console.log(
        chalk.gray(assignment.date),
        '│',
        assignment.chore.padEnd(15),
        '→',
        chalk.yellow(assignment.user)
      );
    });
  }
  
  console.log('');
}

// Fonction pour afficher uniquement les corvées du jour
function showTodayStatus() {
  const assignments = loadJSON(ASSIGNMENTS_PATH);
  const today = getTodayDate();
  const todayAssignments = assignments.filter(a => a.date === today);
  
  if (todayAssignments.length === 0) {
    return;
  }
  
  console.log(chalk.bold.cyan('📋 Statut du jour:\n'));
  
  const completed = todayAssignments.filter(a => a.completed).length;
  const total = todayAssignments.length;
  const percentage = Math.round((completed / total) * 100);
  
  todayAssignments.forEach(assignment => {
    const status = assignment.completed ? chalk.green('✓') : chalk.gray('○');
    console.log(status, chalk.bold(assignment.chore.padEnd(15)), '→', chalk.yellow(assignment.user));
  });
  
  console.log('');
  console.log(chalk.cyan(`Progression: ${completed}/${total} (${percentage}%)`));
  
  if (completed === total) {
    console.log(chalk.green.bold('🎊 Toutes les corvées sont terminées ! Bravo ! 🎊'));
  }
  
  console.log('');
}

module.exports = {
  showStatus,
  showTodayStatus
};

File: chore_manager/commands/status.py
from typing import Any, Dict, List

from ..utils.console import get_console
from ..utils.date_helper import get_today_date
from ..utils.file_manager import ASSIGNMENTS_PATH, CONFIG_PATH, HISTORY_PATH, TOTEM_PATH, load_json
from ..utils.totem_manager import normalize_totem


def show_status() -> Dict[str, Any]:
    console = get_console()
    config = load_json(CONFIG_PATH)
    history = load_json(HISTORY_PATH)
    assignments = load_json(ASSIGNMENTS_PATH)

    console.print("[bold cyan]\n📊 Statut des corvées\n[/]")

    console.print("[bold]Historique total :[/]")
    for user, count in sorted(history.items(), key=lambda item: item[1]):
        bar = "█" * count
        console.print(
            f"[yellow]{user.ljust(30)}[/] [cyan]{bar}[/] [grey70]({count})[/]"
        )

    console.print("")

    today = get_today_date()
    today_assignments = [a for a in assignments if a.get("date") == today]

    if today_assignments:
        console.print(f"[bold]Attributions du jour ({today}) :[/]")
        for assignment in today_assignments:
            status = "[green]✓[/]" if assignment.get("completed") else "[grey70]○[/]"
            console.print(
                f"{status} [bold]{assignment['chore'].ljust(15)}[/] → [yellow]{assignment['user']}[/]"
            )
    else:
        console.print("[grey70]Aucune attribution pour aujourd'hui.[/]")

    totem_raw = load_json(TOTEM_PATH)
    totem = normalize_totem(totem_raw, config["users"])
    console.print("[bold]\nTotem d'immunité :[/]")
    if not totem["immune"]:
        console.print("[grey70]  Immunisés : aucun[/]")
    else:
        console.print("[grey70]  Immunisés :[/] [yellow]" + ", ".join(totem["immune"]) + "[/]")
    if not totem["safe"]:
        console.print("[grey70]  À l'abri : aucun[/]")
    else:
        console.print("[grey70]  À l'abri :[/] [cyan]" + ", ".join(totem["safe"]) + "[/]")
    if not totem["forcedQueue"]:
        console.print("[grey70]  File forcée : aucune[/]")
    else:
        console.print("[grey70]  File forcée :[/] [cyan]" + ", ".join(totem["forcedQueue"]) + "[/]")

    recent_assignments = [a for a in assignments if a.get("date") != today][-5:][::-1]
    if recent_assignments:
        console.print("[bold]\nDernières attributions :[/]")
        for assignment in recent_assignments:
            console.print(
                f"[grey70]{assignment['date']}[/] │ {assignment['chore'].ljust(15)} → [yellow]{assignment['user']}[/]"
            )

    console.print("")

    return {
        "history": history,
        "today_assignments": today_assignments,
        "totem": totem,
        "recent_assignments": recent_assignments,
    }


def show_today_status() -> Dict[str, Any]:
    console = get_console()
    assignments = load_json(ASSIGNMENTS_PATH)
    today = get_today_date()
    today_assignments = [a for a in assignments if a.get("date") == today]

    if not today_assignments:
        return {"today_assignments": []}

    console.print("[bold cyan]📋 Statut du jour:\n[/]")

    completed = len([a for a in today_assignments if a.get("completed")])
    total = len(today_assignments)
    percentage = round((completed / total) * 100)

    for assignment in today_assignments:
        status = "[green]✓[/]" if assignment.get("completed") else "[grey70]○[/]"
        console.print(
            f"{status} [bold]{assignment['chore'].ljust(15)}[/] → [yellow]{assignment['user']}[/]"
        )

    console.print("")
    console.print(f"[cyan]Progression: {completed}/{total} ({percentage}%)[/]")

    if completed == total:
        console.print("[bold green]🎊 Toutes les corvées sont terminées ! Bravo ! 🎊[/]")

    console.print("")

    return {
        "today_assignments": today_assignments,
        "completed": completed,
        "total": total,
        "percentage": percentage,
    }
